swarm.go

```diff
@@ -374,11 +374,21 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 
 	if s.config.SwapEnabled {
-		err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
-		if err != nil {
-			return fmt.Errorf("Unable to deploy swap contract: %v", err)
+		if s.config.Contract != (common.Address{}) {
```

Comment:
I think it would be nicer to extract this into a setup function. too much swap specific logic is here and makes swarm.go harder to read.

Reply:
then you can also eliminate the else-s as you just return

Reply:
Done!

Reply:
@zelig , can you confirm I handled this comment as you intended?